### PR TITLE
Add some technical documentation draft

### PR DESCRIPTION
**Pull request by larkox**
#### Summary
This is a first draft for technical documentation on servicenow.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-32455

### CODE REVIEW

**Review by justinegeffen — COMMENTED** [review]
Thanks, @larkox! Can you confirm where in the docs this content will go? I think it will go well in the README, in addition to the other content. LMKWYT.

**Comment by larkox** [comment]
I think that the readme is not the right place to have the whole document, since it is not "important general information". This is "too techy" and "too deep" for the readme. But I can see the value of a link in the readme to this documentation.

File: docs/technical_documentation.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# ServiceNow app Technical Documentation
+
+## Main functionality
+
+- OAuth connection with one Servicenow instance.
```
Comment (justinegeffen) [inline docs/technical_documentation.md:5]:
```suggestion
- OAuth connection with one ServiceNow instance.
```

File: docs/technical_documentation.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
+# ServiceNow app Technical Documentation
```
Comment (justinegeffen) [inline docs/technical_documentation.md:1]:
```suggestion
# ServiceNow App Technical Documentation
```

File: docs/technical_documentation.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+## Main functionality
+
+- OAuth connection with one Servicenow instance.
+- Creation of tickets on a predefined table (incidents).
+
+## Code overview
+
+- App package:
+  - Contains the business logic of the App. Handles OAuth information, transformation between tables and bindings, and checks on user connection.
```
Comment (justinegeffen) [inline docs/technical_documentation.md:11]:
```suggestion
  - Contains the business logic of the app. Handles OAuth information, transformation between tables and bindings, and checks on user connection.
```

File: docs/technical_documentation.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+- Clients package:
+  - Contains the external clients used by the app. Right now only the ServiceNow client.
+  - It also defines the models used by the client (e.g. Ticket model on ServiceNow side).
+- Config package:
+  - Handles the configuration and its storage. Mainly covers ServiceNow instance and tables (stored in the KV store) and the OAuth config (stored in the OAuth system).
+- Constants package:
+  - Stores all the constant values used by the app.
+- Routers package:
+  - Contain all the routers that accept external input. Right now only covers Mattermost (AWS apps should only receive information from Mattermost).
```
Comment (justinegeffen) [inline docs/technical_documentation.md:20]:
```suggestion
  - Contains all the routers that accept external input. Right now only covers Mattermost (AWS apps should only receive information from Mattermost).
```

File: docs/technical_documentation.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+When the app is initially installed, all users receive the command binding `connect`. The System Admin receives an additional command binding: `onfig OAuth`. Through `config OAuth` we will be able to add the needed information to perform the OAuth connection with ServiceNow: the link to the instance, the client ID, and the client secret.
+
+The OAuth configuration is stored in two different places:
+- The client ID and Secret is stored through the Mattermost OAuth API. 
+- The link to the instance is stored in the KV store along with the configuration.
+
+The connect command generates an ephemeral message with the link to start the OAuth process through Mattermost.
+
+The connect call will:
+- Create the oauth configuration using the information in the context.
```
Comment (justinegeffen) [inline docs/technical_documentation.md:36]:
```suggestion
- Create the OAuth configuration using the information in the context.
```

File: docs/technical_documentation.md [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+- Fetch the state from the call values and fail in case it doesn't exist.
+- Return the AuthCodeURL to Mattermost.
```
Comment (justinegeffen) [inline docs/technical_documentation.md:38]:
```suggestion
- Return the `AuthCodeURL` to Mattermost.
```

File: docs/technical_documentation.md [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+
+The complete call will:
+- Get the code from the call values.
+- Create the oauth configuration using the information in the context.
```
Comment (justinegeffen) [inline docs/technical_documentation.md:42]:
```suggestion
- Create the OAuth configuration using the information in the context.
```

**Review by justinegeffen — COMMENTED** [review]
Second pass review.

**Pull request merged**